qiling/core_hooks.py

```diff
@@ -18,6 +19,73 @@
 from .const import QL_HOOK_BLOCK
 from .exception import QlErrorCoreHook
 
+if TYPE_CHECKING:
```

Comment:
try `from __future__ import annotations`, which is more elegant.

Reply:
importing `annotations` helps with avoiding string annotations for circular dependencies (i.e. no need to to do `ql: 'Qiling'` anymore, just `ql: Qiling`), but it still requires the `TYPE_CHECKING` mechanism.

Is this what you meant..?

Reply:
> importing `annotations` helps with avoiding string annotations for circular dependencies (i.e. no need to to do `ql: 'Qiling'` anymore, just `ql: Qiling`), but it still requires the `TYPE_CHECKING` mechanism.
> 
> Is this what you meant..?

Sorry I missed the notification. Yes you are right, for circular dep, `TYPE_CHECKING` is the only way.